pkg/cloud/cloud.go

```diff
@@ -1,39 +1,68 @@
 package cloud
 
 import (
+	"fmt"
+
 	configv1 "github.com/openshift/api/config/v1"
+	"k8s.io/klog/v2"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud/common"
+	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/config"
+
 	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud/aws"
 	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud/azure"
 	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud/azurestack"
 	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud/ibm"
 	"github.com/openshift/cluster-cloud-controller-manager-operator/pkg/cloud/openstack"
-	"k8s.io/klog/v2"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-// GetResources selectively returns a list resources required for
-// provisioning CCM instance in the cluster for the given platform type
+// GetResources selectively returns a list of resources required for
+// provisioning CCM instance in the cluster for the given OperatorConfig.
 //
 // These resources will be actively maintained by the operator, preventing
 // changes in their spec. However you can extend any resource spec with
 // values not specified in the provided source resource. These changes
 // would be preserved.
-func GetResources(platformStatus *configv1.PlatformStatus) []client.Object {
+func GetResources(operatorConfig config.OperatorConfig) []client.Object {
```

Comment:
Lets hold merge for a while. This need return error, IMO.

Reply:
Why do you think it needs an error? We are handling all of the things that could possibly return an error already in the constructor, this is just returning a pre-rendered value.

I don't want us to over-optimise here just in case this comes up in the future. Changing the signature back in a future PR should be simple enough